src/System.Runtime.Extensions/tests/System/Convert.netcoreapp.cs

```diff
@@ -90,72 +94,258 @@ public void TryToBase64Chars_InvalidOptions_Throws(Base64FormattingOptions inval
         }
 
         [Theory]
-        [InlineData("")]
-        [InlineData("BQYHCA==")]
-        [InlineData(
-            "AAECAwQFBgcICQoLDA0ODxAREhMUFRYXGBkaGxwdHh8gISIjJCUmJygpKissLS4vMDEyMzQ1Njc4\r\n" +
-            "OTo7PD0+P0BBQkNERUZHSElKS0xNTk9QUVJTVFVWV1hZWltcXV5fYGFiY2RlZmdoaWprbG1ub3Bx\r\n" +
-            "cnN0dXZ3")]
-        public void TryFromBase64String_MatchesFromBase64String(string stringInput)
+        [MemberData(nameof(Base64TestData))]
+        public static void TryFromBase64String(string encoded, byte[] expected)
         {
-            byte[] expected = Convert.FromBase64String(stringInput);
-            Span<byte> dest;
+            if (expected == null)
+            {
+                Span<byte> actual = new byte[1000];
+                bool success = Convert.TryFromBase64String(encoded, actual, out int bytesWritten);
```

Comment:
In the case of failed conversion, do we guarantee the contents of the passed in destination span remain unchanged? If so, we should add an assert to validate that.

Reply:
No we don't guarantee that. The implementation can and does scribble over the buffer.